atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -100,6 +102,7 @@ protected SnapshotTransactionManager(
         this.closingCallbacks = new CopyOnWriteArrayList<>();
         this.isClosed = new AtomicBoolean(false);
         this.getRangesExecutor = createGetRangesExecutor(concurrentGetRangesThreadPoolSize);
+        this.timestampTracker = TimestampTracker.createWithDefaultTrackers(timelockService, cleaner);
```

Comment:
This breaks async initialization: we can just register metrics when the `timelockService` and the `cleaner` are up.